packages/store/addon/-private/system/core-store.ts

```diff
@@ -2737,21 +2840,36 @@ class Store extends Service {
       // contains unknown attributes or relationships, log a warning.
 
       if (ENV.DS_WARN_ON_UNKNOWN_KEYS) {
-        let modelClass = this.modelFor(modelName);
+        let unknownAttributes, unknownRelationships;
```

Comment:
this will only work for the case where the schemas for all `type` are identical, not the one where they may differ by identifier.

Reply:
(this is a low priority fix)